jax/_src/random.py

```diff
@@ -1614,15 +1614,17 @@ def orthogonal(
   Returns:
     A random array with the specified shape and dtype.
   """
-  z = normal(key, shape + (n, n), dtype)
+  _check_shape("orthogonal", shape)
+  n = core.concrete_or_error(int, n, "The error occurred in jax.random.orthogonal()")
```

Comment:
Let's use `operator.index` instead of `int` - it will error instead of round if the input is a float.